### PR TITLE
Make compatible with current qiskit nature 0.4.5

### PR DESCRIPTION
**Pull request by LaurinFischer**
### Summary
This PR updates the parsing of Fermionic and Spin operators to the syntax of the current version of qiskit nature (0.4.5)

### List of changes
- replace deprecated `.reduce()` method of `FermionicOp` and `SpinOp` with `.simplify()`
- specify dense labels when converting a `FermionicOp` to a list via `to_list()`
- specify language in the docs configuration
- import aer-related classes directly from qiskit-aer instead through terra.

### CODE REVIEW

File: qiskit_cold_atom/fermions/fermion_simulator_backend.py [hunk 1]
```diff
@@ -20,9 +20,9 @@
 
 from qiskit.providers.models import BackendConfiguration
 from qiskit.providers import Options
-from qiskit.providers.aer import AerJob
 from qiskit import QuantumCircuit
 from qiskit.result import Result
+from qiskit_aer import AerJob
```
Comment (eggerdj) [inline qiskit_cold_atom/fermions/fermion_simulator_backend.py:25]: Why the change? I though `qiskit.providers.aer` was the right import location.

File: test/test_fermion_simulator_backend.py [hunk 2]
```diff
@@ -15,9 +15,9 @@
 from time import sleep
 import numpy as np
 
+from qiskit_aer import AerJob
```
Comment (eggerdj) [inline test/test_fermion_simulator_backend.py:18]: See question above on the import.

**Review by eggerdj — COMMENTED** [review]
Thanks a lot. This looks good. So the main changes are that `reduce` has been changed to `simplify`. Why did the aer import need changing?

File: qiskit_cold_atom/fermions/fermion_simulator_backend.py (hunk 1 above)
Comment (LaurinFischer) [reply qiskit_cold_atom/fermions/fermion_simulator_backend.py:25]: My pylint was complaining about the old import and when calling `from qiskit import Aer`, a `PendingDeprecationWarning` triggers that recommends to import from the qiskit-aer package directly, so I thought it made sense to do this for all aer-related imports.

**Review by eggerdj — APPROVED** [review]
LGTM
